File: flutter/lib/data/models/medication.dart
class Medication {
  final int id;
  final String name;
  final String dose;
  final int frequencyPerDay;
  final DateTime startDate;
  final DateTime endDate;
  final List<MedicationDoseSchedule> doseSchedules;

  Medication({
    required this.id,
    required this.name,
    required this.dose,
    required this.frequencyPerDay,
    required this.startDate,
    required this.endDate,
    required this.doseSchedules,
  });

  factory Medication.fromJson(Map<String, dynamic> json) {
    return Medication(
      id: json['id'],
      name: json['name'] ?? 'Bilinmeyen İlaç',
      dose: json['dose'] ?? '',
      frequencyPerDay: json['frequencyPerDay'] ?? 0,
      startDate: DateTime.tryParse(json['startDate'] ?? '') ?? DateTime.now(),
      endDate: DateTime.tryParse(json['endDate'] ?? '') ?? DateTime.now(),
      doseSchedules: (json['doseSchedules'] as List?)
              ?.map((i) => MedicationDoseSchedule.fromJson(i))
              .toList() ?? [],
    );
  }
}

class MedicationDoseSchedule {
  final int id;
  final DateTime scheduledTime;
  final String? notes; 
  final bool isTaken;

  MedicationDoseSchedule({
    required this.id,
    required this.scheduledTime,
    this.notes, 
    this.isTaken = false,
  });

  factory MedicationDoseSchedule.fromJson(Map<String, dynamic> json) {
    
    bool takenStatus = false;
    if (json['records'] != null && (json['records'] as List).isNotEmpty) {
      takenStatus = (json['records'] as List).any((r) => r['isTaken'] == true);
    }

    return MedicationDoseSchedule(
      id: json['id'],
      scheduledTime: DateTime.tryParse(json['scheduledTime'] ?? '') ?? DateTime.now(),
      notes: json['notes'], 
      isTaken: takenStatus,
    );
  }
}

File: backend/Models/Medication.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace health.api.Models
{
    public class Medication
    {
        public int Id { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public int PatientId { get; set; }
        public Patient? Patient { get; set; } // ← BURASI DEĞİŞTİ

        [Required, MaxLength(100)]
        public string Name { get; set; }

        [Required, MaxLength(50)]
        public string Dose { get; set; }

        public int FrequencyPerDay { get; set; }
        public int DurationDays { get; set; }

        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }

        [MaxLength(255)]
        public string? Notes { get; set; }

        public ICollection<MedicationDoseSchedule>? DoseSchedules { get; set; }
        public ICollection<MedicationRecord>? MedicationRecords { get; set; }
        public ICollection<SideEffect>? SideEffects { get; set; }
    }
}
